File: spec/javascript/test_debug_array_user.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('DebugArrayUser', 'src/fixed_struct.bin', function(r, DebugArrayUser_) {
  // --debug implies --no-auto-read
  r._read();

  assert.strictEqual(r.oneCat.meow, 80);
  assert.strictEqual(r.arrayOfCats.length, 3);
  assert.strictEqual(r.arrayOfCats[0].meow, 65);
  assert.strictEqual(r.arrayOfCats[1].meow, 67);
  assert.strictEqual(r.arrayOfCats[2].meow, 75);
});

File: spec/lua/test_debug_array_user.lua
local luaunit = require("luaunit")

require("debug_array_user")

TestDebugArrayUser = {}

function TestDebugArrayUser:test_debug_array_user()
    local r = DebugArrayUser:from_file("src/fixed_struct.bin")

    -- --debug implies --no-auto-read
    r:_read()

    luaunit.assertEquals(r.one_cat.meow, 80)
    luaunit.assertEquals(#r.array_of_cats, 3)
    luaunit.assertEquals(r.array_of_cats[0 + 1].meow, 65)
    luaunit.assertEquals(r.array_of_cats[1 + 1].meow, 67)
    luaunit.assertEquals(r.array_of_cats[2 + 1].meow, 75)
end
